docs/optimize-sensors.md

```diff
@@ -0,0 +1,55 @@
+## Introduction
+
+We have been optimising all parts of the analysis as part of our effort to reduce the total scan times. Some of those optimizations depend on additional information exchanged
+between analyzers and the platform. Here we list some of the APIs introduced specifically to allow certain optimizations to be performed. Note the implementation of these APIs are
+optional.
+
+## Analyzer cache
+
+Analyzers can persist data in a cache that will be made available to it in a later analysis. The cached data is stored on the server side and the analyzer can store any key.
+Analyzers must make sure that there’s no risk of key collision between different analyzers.
+
+Note that cached data doesn’t necessarily come from the previous analysis of the same branch. In the current SQ’s implementation:
+
+| Case | Origin of the cache
+| --- |---------------------|
+| 1st analysis of a branch | Main branch         |
```

Comment:
this contradicts what's in the description of https://sonarsource.atlassian.net/browse/SONAR-16187
it's not always the main branch

Reply:
I can't spot the difference. If the branch (not PR) doesn't exist, it uses the main branch.

Reply:
@sns-seb I'll merge it, please let me know if you still think there's a inconsistency so that we can tackle it.

Reply:
reread the ticket and the doc, they actually match, sorry for the noise

the way it's put in the doc mislead me: I read "branch" as a generic term for both "branch and PR" and missed the last line in the table about PR